magento/pwa-studio
# [feature]: Order confirmation Page

## Description

The order confirmation page displays data from the order that was just placed. It's different from most of our other recent components in that it must render "past" data that is no longer stored/cached.

I approached this by prefixing a query on the submission handler that fetches _current_ order details right before placing the order. The details are then passed to the order confirmation component for rendering. The only queries/mutations that exist within the order confirmation page and its children are those around creating an account.

**Note:**

We combined the subscribe and create account forms into a single form due to restrictions from graphql.

Refer a friend is also deferred.

## Related Issue





Closes [PWA-186](https://jira.corp.magento.com/browse/PWA-186).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. As a guest, place an order. See that the order confirmation page appears.
2. Create an account using the form. It should log you in but not redirect you for now. Eventually this will redirect to an account view or orders page.
3. As an authed user, place an order. See that the order confirmation page appears but there is no  create account form.

If shipping info/method/payment info sections are not merged you may need to manually submit this information to the cart through graphql queries. I'll provide some that should work, just replace cart id with your own from the store. 

**To add shipping address/method**

Go to the `/cart` page and estimate your shipping. That will attach a fake address as well as select a shipping method for your cart.

**To add payment method:**

```gql
mutation {
    setPaymentMethodOnCart(input: {
        cart_id: "e3ih8dF98xpiykcHkHqb9XSsppOSymMK"
        payment_method: {
            code: "checkmo"
        }
    }) {
        cart {
        selected_payment_method {
            code
        }
        }
    }
}
```

**To add billing address:**
```gql
mutation {
  setBillingAddressOnCart(
    input: {
      cart_id: "e3ih8dF98xpiykcHkHqb9XSsppOSymMK"
      billing_address: {
        address: {
          firstname: "John"
          lastname: "Doe"
          company: "Company Name"
          street: ["320 N Crescent Dr", "Beverly Hills"]
          city: "Los Angeles"
          region: "CA"
          postcode: "90210"
          country_code: "US"
          telephone: "123-456-0000"
          save_in_address_book: false
        }
      }
    }
  ) {
    cart {
      billing_address {
        firstname
        lastname
        company
        street
        city
        region{
          code
          label
        }
        postcode
        telephone
        country {
          code
          label
        }
      }
    }
  }
}
```

## Screenshots / Screen Captures (if appropriate)

[![Image from Gyazo](https://i.gyazo.com/54f108a2bb064c3ebd9ea0427d0e6d45.png)](https://gyazo.com/54f108a2bb064c3ebd9ea0427d0e6d45)

[![Image from Gyazo](https://i.gyazo.com/aa1d347742af044a5fe1f526cc7a5394.png)](https://gyazo.com/aa1d347742af044a5fe1f526cc7a5394)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
